mantle/cmd/ore/gcloud/update-image.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/coreos/mantle/platform/api/gcloud"
+)
+
+var (
+	cmdUpdate = &cobra.Command{
+		Use:   "update-image --image=ImageName [--family=ImageFamily] [--description=ImageDescription]",
+		Short: "Update os image",
+		Long:  "Update os image attributes in GCP.",
+		Run:   runUpdateImage,
+	}
+
+	updateImageName        string
+	updateImageFamily      string
+	updateImageDescription string
+)
+
+func init() {
+	cmdUpdate.Flags().StringVar(
+		&updateImageName, "image", "", "GCP image name")
+	cmdUpdate.Flags().StringVar(
+		&updateImageFamily, "family", "",
+		"Updated GCP image family to attach image to")
+	cmdUpdate.Flags().StringVar(
+		&updateImageDescription, "description", "",
+		"The updated description for the image")
+	GCloud.AddCommand(cmdUpdate)
+}
+
+func runUpdateImage(cmd *cobra.Command, args []string) {
+	// Check that the user provided an image
+	if updateImageName == "" {
+		plog.Fatal("Must provide an image name via --image")
+	}
+	// Check that the user provided at least one thing to change
+	if updateImageFamily == "" && updateImageDescription == "" {
+		plog.Fatal("Must provide one of --family or --description")
+	}
+
+	// Create the ImageSpec. Don't worry about passing "" for
+	// family or description. If "" is passed no update will happen.
+	spec := &gcloud.ImageSpec{
+		Name:        updateImageName,
+		Family:      updateImageFamily,
+		Description: updateImageDescription,
+	}
+
+	_, err := api.UpdateImage(spec)
```

Comment:
Is this alpha API open to all accounts?

If not does the endpoint this is hitting have a cleanly understandable error case when an account is not flagged?

Reply:
It's open to whitelisted projects. I know when using the gcloud CLI it gave me a clear error message but I'm not sure I tried it with `ore` until after all of our projects were whitelisted. I'll have to create a new project to test this, unless someone else would like to try it on a CL image in the CL project and report.

Reply:
ok I didn't have a good answer for you until right now. Yesterday I went through and created a new project (i.e. not whitelisted) so i could test this and the command succeeded just fine (except no image was created). Which was obviously an error.

We were't properly catching errors from this API call so I opened a PR to fix that: https://github.com/coreos/coreos-assembler/pull/1454

With the changes in that PR it now shows:

```
buildextend-gcp --upload --project dusty-project --bucket gs://20200513-dusty-project-image-uploads/image-import --js
on /srv/20200513-dusty-project-gcp-image-upload.json --create-image=true --license=fedora-coreos-testing --family=dusty --deprecated
[INFO]: symlink is for a build and publish command
[INFO]: ore target gcp found via symlink
[INFO]: Targeting build: 31.20200513.dev.1
[INFO]: Processed build for: Fedora CoreOS testing (FEDORA-COREOS-x86_64) 31.20200513.dev.1
[INFO]: operating on fedora-coreos-31.20200513.dev.1-gcp.x86_64.tar.gz
[INFO]: executing upload commands for ore
+ ore gcloud --project dusty-project --json-key /srv/20200513-dusty-project-gcp-image-upload.json upload --basename fedora-coreos --bucket gs://20200513-dusty-project-image-uploads/image-import --name fedora-coreos-31-20200513-dev-1-gcp-x86-64 --file builds/31.20200513.dev.1/x86_64/fedora-coreos-31.20200513.dev.1-gcp.x86_64.tar.gz --writ
e-url /srv/tmp/tmpgwgjh5qr/gcp-url --license fedora-coreos-testing
File image-import/fedora-coreos-31-20200513-dev-1-gcp-x86-64.tar.gz already exists on Google Storage. Overwrite? (y/n):n
Skipped file upload
Creating image in GCE: fedora-coreos-31-20200513-dev-1-gcp-x86-64...
Error listing images in GCE: Listing GCE images failed: googleapi: Error 403: Required 'Alpha Access' permission for 'Compute API', forbidden
Error running command ore
+ rc=1
+ set +x

```


I think the `Error 403: Required 'Alpha Access' permission for 'Compute API', forbidden` should suffice for what you were looking for.

Reply:
Yep that's perfect.

Reply:
ok I'm leaving the error logic to be fixed in https://github.com/coreos/coreos-assembler/pull/1454 and won't worry about it in this PR

Reply:
#1454 merged so marking this as resolved.